schemas/JSON/manifests/v1.2.0/manifest.installer.1.2.0.json

```diff
@@ -449,37 +480,33 @@
           "$ref": "#/definitions/MinimumOSVersion"
         },
         "Architecture": {
-          "type": "string",
-          "enum": [
-            "x86",
-            "x64",
-            "arm",
-            "arm64",
-            "neutral"
-          ],
-          "description": "The installer target architecture"
+          "$ref": "#/definitions/Architecture"
         },
         "InstallerType": {
           "$ref": "#/definitions/InstallerType"
         },
+        "ArchiveInstallers": {

```

Comment:
Since we do decide to use multiple ArchiveInstallers, do we have decision on the installer selection behaviors? (i.e. same app with different installers vs different apps)

For example, for some apps, we may have below and we want to install both of them
app1.exe
app2.exe

For some apps, we may have below and we want to only pick 1 to install
app1_x86.exe
app1_x64.exe

I think we can only support 1 of the scenarios above without further manifest entries and making the logic more complex.

Reply:
After discussing this further, we're leaning towards supporting the following scenarios for zip:

1. Single or multiple portable exes contained inside a zip, which will all be installed.
2. Single Installer inside a zip.
3. Multiple installers inside a zip, but only installing one from the selection of installers. This would ensure that we support examples such as zips having multiple architecture installers or multiple locale installers residing in the same zip. If a specific architecture or locale is not specified, winget will choose the best option to install for you.

Regarding the edge case of a zip file containing multiple installerTypes (such as msi and exe), we would need to add support for enabling a user to select which installerType they would prefer to install.

Reply:
I do not support this:
> multiple portable exes contained inside a zip, which will all be installed.

That will just complicate things unnecessarily (although given how messy things are even when we try not to be, it might actually help us to need to handle it head on).  But generally, dependencies are a better solution than having a bunch of stuff in a single archive and installing it all.  Even if all of those things are in a single archive.

Option 3 is just like our current manifest, but directly supporting the installer set expansion from a single archive rather than requiring repetition of the archive as a top level installer.

Reply:
Updated the comment above. We decided that we will only install a single portable inside a zip even if the zip archive contains multiple portables. This is to ensure that we limit the complexity that comes with installing multiple packages at the same time. It is better to have that be driven through dependencies when that feature is ready.

Reply:
why not just select the main portable exe, chuck a shortcut to the desktop and the folder to program files?